app/javascript/controllers/partner_filter_component_controller.js

```diff
@@ -0,0 +1,71 @@
+import { Controller } from "@hotwired/stimulus";
+
+// Connects to data-controller="partner-filter-component"
+export default class extends Controller {
+	static targets = [
+		"form",
+		"category",
+		"categoryText",
+		"categoryDropdown",
+		"neighbourhood",
+		"neighbourhoodText",
+		"neighbourhoodDropdown",
+	];
+
+	connect() {
+		this.updateLabels();
+	}
+
+	submitCategory() {
+		this.submitForm();
+	}
+
+	submitNeighbourhood() {
+		this.submitForm();
+	}
+
+	resetCategory() {
+		this.selectedCategory.checked = false;
+		this.submitForm();
+	}
+
+	resetNeighbourhood() {
+		this.selectedNeighbourhood.checked = false;
+		this.submitForm();
+	}
+
+	toggleCategory() {
```

Comment:
this should be spun into its own stimulus controller but waiting on sorting the components directory out first